### DISCUSSION TITLE
No obvious way of overriding trustStore in flatpak installation or importing a cert into jre cacerts

### DISCUSSION THREAD

**Discussion by sochotnicky** [dbeaver/dbeaver#16745] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version: flatpak (Linux/Gentoo)
- DBeaver version: 22.0.2
- Additional extensions: None

#### Connection specification:
- Need to override SSL CA for trino connection using internal SSL CA

#### Describe the problem you're observing:

I am trying to connect to internal trino database which is using company internal CA. Normally I'd do either of the following:

- Import the certificate into JAVA truststore (jre/lib/security/cacerts) - this is impossoble because in flatpak the file is read-only
- Add custom -Djavax.net.ssl.trustStore and point to my personal certstore (as described in https://github.com/dbeaver/dbeaver/issues/4341). But the override is done in `dbeaver.ini` which in flatpak resides in '/app/dbeaver/dbeaver.ini' which is read-only

Since neither step can be done in flatpak I am not quite sure what way there is for me to import custom CAs into java trustStore that would be picked up by dbeaver.

Downloaded zip bundle (not flatpak) works fine after the bundled cacert is updated.

#### Include any warning/errors/backtraces from the logs

Without the custom CA I just end up with following error:
```
unable to find valid certification path to requested target
```

**Comment by flambchap** [comment]
Hi @sochotnicky. Thank you for the report. We are working on this ticket.

**Comment by riednyko** [comment, marked as answer]
Hi @sochotnicky. You can set property `-Djavax.net.ssl.truststore` with your custom path to your cert in **config.ini** file if **dbeaver.ini** is read-only. [This documentation](https://github.com/dbeaver/dbeaver/wiki/How-to-set-variable-if-dbeaver.ini-is-read-only) will help you.

**Comment by sochotnicky** [comment]
I suppose that should work.I missed the docs - I think it might be a nice addition to tweak/add to https://github.com/dbeaver/dbeaver/wiki/Importing-CA-Certificates-into-DBeaver and cross link the other wiki page too?
